classes/s3/helper.php

```diff
@@ -0,0 +1,139 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_lcbackupcoursestep\s3;
+
+use Aws\MockHandler;
+use Aws\Result;
+use stdClass;
+use stored_file;
+use tool_lifecycle\local\manager\settings_manager;
+use tool_lifecycle\settings_type;
+
+/**
+ * Helper class to work with s3 bucket.
+ *
+ * @package     tool_lcbackupcoursestep
+ * @copyright   2024 Catalyst IT
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+
+    /**
+     * Create a new S3 client.
+     *
+     * @param array $settings amazon s3 settings.
+     * @return \Aws\S3\S3Client
+     */
+    public static function create_client($settings) {
+        // Connection options.
+        $options = [
+            'version' => 'latest',
+            'region' => $settings['s3_region'] ?? 'ap-southeast-2',
+        ];
+
+        // Credentials.
+        $usesdkcreds = $settings['s3_usesdkcreds'] ?? false;
+        if (!$usesdkcreds) {
+            $options['credentials'] = [
+                'key' => $settings['s3_key'],
+                'secret' => $settings['s3_secret'],
+            ];
+        }
+
+        // Proxy.
+        if ($settings['s3_useproxy']) {
+            $options['http'] = ['proxy' => \local_aws\local\aws_helper::get_proxy_string()];
+        }
+
+        // Test only.
+        if (PHPUNIT_TEST) {
+            $mock = new MockHandler();
+            $mock->append(new Result([]));
+            $options['handler'] = $mock;
+        }
+
+        return new \Aws\S3\S3Client($options);
+    }
+
+    /**
+     * Test connection to S3 bucket.
+     *
+     * @param array $settings amazon s3 settings.
+     * @return stdClass
+     */
+    public static function check_connection($settings): stdClass {
+        $connection = new stdClass();
+        $connection->success = true;
+        $connection->details = '';
+        try {
+            $client = self::create_client($settings);
+            $client->headBucket(['Bucket' => $settings['s3_bucket']]);
+        } catch (\Aws\S3\Exception\S3Exception $e) {
+            $connection->success = false;
+            $connection->details = $e->getMessage();
+        } catch (\Exception $e) {
+            $connection->success = false;
+            $connection->details = $e->getMessage();
+        }
```

Comment:
What is the point to have two catches here with the same bodies?

Reply:
I have combined the catch